packages/core/useFetch/index.ts

```diff
@@ -235,15 +241,25 @@ export function useFetch<T>(url: MaybeRef<string>, ...args: any[]): UseFetchRetu
       }
     }
 
+    let _url = unref(url)
+    let _fetchOptions = fetchOptions
+
+    if (options.beforeFetch) {
+      const { url: maybeUrl, options: maybeOptions } = await options.beforeFetch({ url: _url, options: _fetchOptions }, abort)
+
+      _url = maybeUrl || _url
+      _fetchOptions = maybeOptions || _fetchOptions
+    }
```

Comment:
```suggestion
    let context = { url: unref(url), options: fetchOptions, cancel: abort }

    if (options.beforeFetch)
      Object.assign(context, await options.beforeFetch(context))
```

Reply:
Would `abort` work before using fetch? So should we handle it with another approach (like not fire-up fetch at all)?

Reply:
Oh haha yeah my bad 😂 

Perhaps we could also allow them to return a boolean? So they can return context, which would modify the request, `false`, which would not execute the request, and `true` which would execute the request without modifying the parameters. Does that sound okay to you?

```ts
beforeFetch?: (ctx: Context) => Promise<Context | boolean> | Context | boolean
```

Reply:
My initial thought is to return boolean as well, but I am a little concerned about if the intention is clear enough. I was confused about [`return false` in event handlers](https://stackoverflow.com/questions/1357118/event-preventdefault-vs-return-false) for a long time, not sure if it means "cancel the event" or "cancel the handler" or just the "result".

So I guess it might be more clear and explicit to call `cancel()` or `abort()` in this case.

Reply:
That's a good point. So we should go with your way then. As for what we call it I think we should go with `cancel()`(which you also had in your commit). Using `abort()` might cause confusion since the function already exports an `abort()` method and people may accidentally do something like

```ts
const { abort } = useFetch('', {
  beforeFetch({ options }) {
    abort()

    return {
      options,
    }
  },
})
```